src/lib/viewers/doc/BoxExcel/charts.js

```diff
@@ -0,0 +1,237 @@
+import React, { Component } from 'react';
+import { Bar, Pie, Doughnut } from 'react-chartjs-2';
+import PropTypes from 'prop-types';
+import { HEADER_WIDTH, HEADER_HEIGHT } from './const';
+
+/**
+ * map chart type to React Component imported from 'react-chartjs-2'
+ *
+ * @type {Object}
+ */
+const chartTypeMap = {
+    line: Bar,
+    bar: Bar,
+    pie: Pie,
+    doughnut: Doughnut
+};
+
+/**
+ * map legend to chartjs-recognizable
+ *
+ * @type {Object}
+ */
+const legendMap = {
+    b: 'bottom'
+};
+
+class Charts extends Component {
+    /* eslint-disable */
+    /**
+     * sheet spreadsheet data
+     * themeColors array of colors
+     *
+     * @type {Object}
+     */
+    static propTypes = {
+        sheet: PropTypes.object.isRequired,
+        zoom: PropTypes.number,
+        themeColors: PropTypes.array.isRequired,
+        scrollLeftOffset: PropTypes.number.isRequired,
+        scrollTopOffset: PropTypes.number.isRequired
+    };
+    /* eslint-enable */
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} props [description]
+     */
+    constructor(props) {
+        super(props);
+
+        const { sheet } = this.props;
+        this.state = {
+            charts: sheet['!charts'].map(this._parseChart)
+        };
+    }
+
+    /**
+     * Helper function of _parseChart
+     * Re-organize the data of a subplot
+     *
+     * @param  {Object} chart   Chart object parsed from sheetjs
+     * @param  {Array} subPlot  'cat' means chart label, 'val' means chart data
+     * @param  {number} counter counter of the whole chart
+     * @return {Object}         labels and data of a subplot, and updated counter
+     */
+    _getRawData = (chart, subPlot, counter) => {
+        /* global XLSX */
+        const { e } = XLSX.utils.decode_range(chart['!ref']);
+        const rowCount = e.r + 1;
+        let subLabels = [];
+        const subData = [];
+        let newCounter = counter;
+        subPlot.forEach((key) => {
```

Comment:
Might be worth moving this parsing method into a util function so it's abstracted away and makes the code easier to understand. Something like 
`subPlot.forEach((key) => getSubPlotValue(chart, newCounter))` or something similar that makes sense